pkgs/applications/networking/cluster/kuma/default.nix

```diff
@@ -1,19 +1,41 @@
-{ lib, fetchFromGitHub, buildGoModule }:
+{ lib
+, fetchFromGitHub
+, buildGoModule
+, coredns
+, onlyCli ? false
+, tags ? [
+    "gateway"
+  ]
+, components ? [
+    "kuma-cp"
+    "kuma-prometheus-sd"
+    "kuma-dp"
+  ]
+}:
 
 buildGoModule rec {
-  pname = "kumactl";
+  pname = "kuma";
   version = "1.3.1";
 
   src = fetchFromGitHub {
     owner = "kumahq";
     repo = "kuma";
     rev = version;
+    leaveDotGit = true;
     sha256 = "0b554cngg2j3wnadpqwhq3dv3la8vvvzyww2diw4il4gl4j6xj0j";
   };
 
   vendorSha256 = "0r26h4vp11wbl7nk3y7c22p60q7lspy8nr58khxyczdqjk6wrdjp";
 
-  subPackages = [ "app/kumactl" ];
+  nativeBuildInputs = (if onlyCli then [] else [ coredns ]);
```

Comment:
Please use `lib.optional` for this check

Reply:
Many thank's for review, i changed it!